application/src/main/java/org/togetherjava/tjbot/commands/moderation/AuditCommand.java

```diff
@@ -139,26 +138,58 @@ private boolean handleChecks(@NotNull Member bot, @NotNull Member author,
         return ModerationUtils.handleHasAuthorRole(ACTION_VERB, hasRequiredRole, author, event);
     }
 
-    private void auditUser(@NotNull User user, @NotNull ISnowflake guild,
-            @NotNull IReplyCallback event) {
-        List<ActionRecord> actions =
-                actionsStore.getActionsByTargetAscending(guild.getIdLong(), user.getIdLong());
+    private Message auditMessage(long guildId, long targetId, int pageNo, @NotNull JDA jda) {
+        List<ActionRecord> actions = actionsStore.getActionsByTargetAscending(guildId, targetId);
 
-        MessageEmbed summary = createSummaryMessage(user, actions);
-        if (actions.isEmpty()) {
-            event.replyEmbeds(summary).queue();
-            return;
+        List<List<ActionRecord>> groupedActions = new ArrayList<>();
+        for (int i = 0; i < actions.size(); i++) {
+            if (i % 25 == 0) {
+                groupedActions.add(new ArrayList<>(25));
+            }
+
+            groupedActions.get(groupedActions.size() - 1).add(actions.get(i));
+        }
+
+        int totalPage = groupedActions.size();
+
+        EmbedBuilder audit = createSummaryEmbed(jda.retrieveUserById(targetId).complete(), actions)
+            .setFooter("Page: " + pageNo + "/" + totalPage);
+        groupedActions.get(pageNo - 1)
+            .forEach(action -> audit.addField(actionToField(action, jda)));
+
+        return new MessageBuilder(audit.build())
+            .setActionRows(ActionRow.of(previousButton(guildId, targetId, pageNo),
+                    nextButton(guildId, targetId, pageNo, totalPage)))
+            .build();
+    }
+
+    private Button previousButton(long guildId, long targetId, int pageNo) {
+        Button previousButton = Button.primary(generateComponentId(String.valueOf(guildId),
+                String.valueOf(targetId), String.valueOf(pageNo), "-1"), "⬅");
+
+        if (pageNo == 1) {
+            previousButton = previousButton.asDisabled();
+        }
+
+        return previousButton;
+    }
+
+    private Button nextButton(long guildId, long targetId, int pageNo, int totalPage) {
+        Button nextButton = Button.primary(generateComponentId(String.valueOf(guildId),
+                String.valueOf(targetId), String.valueOf(pageNo), "1"), "➡");
```

Comment:
Prefer putting the button creation itself into a helper method instead of duplicating it:
```java
private Button createButton(String label, ...) {
  return Button.primary(...);
}
```